src/colmap/estimators/bundle_adjustment.h

```diff
@@ -270,6 +282,46 @@ class RigBundleAdjuster : public BundleAdjuster {
   std::unordered_set<double*> parameterized_quats_;
 };
 
+class PositionPriorBundleAdjuster : public BundleAdjuster {
+ public:
+  struct Options {
+    // Whether to use prior camera positions
+    bool use_prior_position = true;
+
+    // Standard deviation on the position priors
+    Eigen::Matrix3d prior_position_covariance = Eigen::Matrix3d::Identity();
+  };
+
+  PositionPriorBundleAdjuster(const BundleAdjustmentOptions& options,
+                              const BundleAdjustmentConfig& config);
+
+  PositionPriorBundleAdjuster(const BundleAdjustmentOptions& options,
+                              const Options& prior_options,
+                              const BundleAdjustmentConfig& config);
+
+  bool Solve(Reconstruction* reconstruction);
+
+  // Set up the problem
+  void SetUpProblem(Reconstruction* reconstruction,
+                    ceres::LossFunction* loss_function,
+                    ceres::LossFunction* prior_loss_function);
+
+ private:
+  void AddImageToProblem(image_t image_id,
+                         Reconstruction* reconstruction,
+                         ceres::LossFunction* loss_function,
+                         ceres::LossFunction* prior_loss_function);
+
+  Options prior_options_;
+
+  std::unique_ptr<ceres::LossFunction> prior_loss_function_;
+
+  // Rotation from camera frame to world frame (i.e. prior pos frame)
+  // to be used to allow the alignment w.r.t. to the position priors
+  // through a common rotation even when some poses are set as const
+  Eigen::Vector4d qwc0_ = Eigen::Vector4d(1.0, 0.0, 0.0, 0.0);
```

Comment:
This seems to not be used anywhere?

Reply:
You are right.  This part is not used right now as prior positions are only used in the global BA with no constant pose.  I just removed it.